mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/ClassLoaderHandle.java

```diff
@@ -47,6 +47,8 @@ public interface ClassLoaderHandle extends Closeable {
     UserCodeClassLoader getOrResolveClassLoader(Collection<URI> requiredJarFiles,
                                                 Collection<URL> requiredClasspaths) throws IOException;
 
+    void cacheJobArtifacts(Collection<URI> artifacts);
```

Comment:
for your consideration: this API seems a bit awkward to live in classloaderHandle as it has no relation to any class loader? Maybe move to its own class?

Reply:
good point. I'm leaving this to a later refactor on our internal `DefaultClassLoaderHandle` where I want to separate it from handling the blobstore. After that I can move this functionality into the new blob store manager.